pkg/vault/contracts/VaultFactory.sol

```diff
@@ -73,9 +80,13 @@ contract VaultFactory is Authentication {
         bytes32 salt,
         address targetAddress,
         bytes calldata vaultCreationCode,
-        bytes calldata vaultAdminCreationCode,
-        bytes calldata vaultExtensionCreationCode
-    ) external authenticate {
+        bytes calldata vaultExtensionCreationCode,
+        bytes calldata vaultAdminCreationCode
+    ) external onlyOwner nonReentrant {
+        if (deploymentComplete) {
```

Comment:
It seems that this is not necessary because on re-entry it will crash on create2 and nonReentrant the same

Reply:
It does seem like create2 would revert the second time, because we're hard-coding the salt to 0. So if it deployed once, then you tried it again with the same salt, it would fail. We'd need a counter for the salt or something to allow it to run multiple times with create2.

Reply:
it will only revert on create2 (`CreateCollision`) if passed the same salt.

my understanding is that this boolean thus prevents the factory from deploying more than once